JotunnLib/Extensions/GameObjectExtension.cs

```diff
@@ -104,6 +106,174 @@ public static GameObject OrNull(this GameObject @this)
 
             return target;
         }
+
+        /// <summary>
+        ///     Extension method to check if GameObject has a component.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="gameObject"></param>
+        /// <returns></returns>
+        public static bool HasComponent<T>(this GameObject gameObject) where T : Component
+        {
+            return gameObject.GetComponent<T>() != null;
```

Comment:
Unity actually discourages doing null checks against Unity objects (https://docs.unity3d.com/Manual/BestPracticeUnderstandingPerformanceInUnity7.html at Null comparisons against UnityEngine.Object subclasses)

Thus, the method should be simplefied to `return gameObject.GetComponent<T>()`, at which point the whole method is rather redundent and doesn't have a purpose anymore. It should be removed in my opinion, same for the string overload.

Reply:
Oh, that's very good to know. Thank you! Will remove that then.

Reply:
And also update some of my own code in other projects now that I know that